flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -810,6 +817,7 @@ else if (transitionState(current, ExecutionState.FAILED, t)) {
 					}
 					// else fall through the loop and
 				}
+				waitInvokableCancelCompletion();
```

Comment:
Why is it placed only in the `catch(...)` block? Don't we want to wait in case of clean shutdown or clean completion?

Reply:
Yes, we do! I'll fix it, thanks.